src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -422,8 +422,13 @@ private static void MarkTransitiveOrigin(Dictionary<string, TransitiveEntry> tra
         internal static TransitivePackageReference MergeTransitiveOrigin(PackageReference currentPackage, TransitiveEntry transitiveEntry)
         {
             var transitiveOrigins = new SortedSet<PackageReference>(GetPackageReferenceUtility.PackageReferenceMergeComparer);
-
-            transitiveEntry?.Keys?.ForEach(key => transitiveOrigins.AddRange(transitiveEntry[key]));
+            transitiveEntry?.Keys?.ForEach(key =>
```

Comment:
I don't think Keys can ever be null in a dictionary.

Reply:
Removed `?` in `.Keys` accessor